### PR TITLE
[CALCITE-6315] Support PostgreSQL TO_CHAR, TO_DATE, TO_TIMESTAMP

### PR DESCRIPTION
**Pull request by normanj-bitquill**
* Both functions use PostgreSQL format patterns
* Added tests for format patterns supported by PostgreSQL but missing from Calcite
* If the data or timestamp cannot be parsed using format string, then an exception is thrown.

### CODE REVIEW

**Comment by mihaibudiu** [comment]
The style checker has some complaints about your code. You can see them in the failed CI run.

**Comment by normanj-bitquill** [comment]
@mihaibudiu Thank you, I have addressed the style checker error. The latest CI run failed when trying to pull the redis docker image.

File: testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java [hunk 1]
```diff
@@ -4465,44 +4465,35 @@ void testBitGetFunc(SqlOperatorFixture f, String functionName) {
     f.checkString("to_char(timestamp '2022-06-03 12:15:48.678', 'YYYY-MM-DD HH24:MI:SS.MS TZ')",
         "2022-06-03 12:15:48.678",
         "VARCHAR NOT NULL");
-    f.checkString("to_char(timestamp '2022-06-03 12:15:48.678', 'Day')",
```
Comment (mihaibudiu) [inline testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java:4468]: why delete these tests?

File: testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java [hunk 2]
```diff
@@ -4465,44 +4465,35 @@ void testBitGetFunc(SqlOperatorFixture f, String functionName) {
...
+  }
+
+  @Test void testToTimestamp() {
+    final SqlOperatorFixture f = fixture().withLibrary(SqlLibrary.POSTGRESQL);
+    f.setFor(SqlLibraryOperators.TO_TIMESTAMP);
+
+    f.checkString("to_timestamp('2022-06-03 18:34:56', 'YYYY-MM-DD HH24:MI:SS')",
+        "2022-06-03 18:34:56",
+        "TIMESTAMP(0) NOT NULL");
+    f.checkFails("to_timestamp('ABCD', 'YYYY-MM-DD HH24:MI:SS')",
```
Comment (mihaibudiu) [inline testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java:4492]: please add more tests, including nulls, and illegal value for date, and dates before the Gregorian calendar

**Review by mihaibudiu — COMMENTED** [review]
I don't understand what happens to TO_CHAR in this PR.

**Comment by normanj-bitquill** [comment]
> I don't understand what happens to TO_CHAR in this PR.

@mihaibudiu Good point. I have updated the title of this PR and the commit message to say that I only added support for `TO_DATE` and `TO_TIMESTAMP`.

File: testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java (hunk 1 above)
Comment (normanj-bitquill) [reply testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java]: A mistake on my part. They have been added back, and I have added some more tests.

File: testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java (hunk 2 above)
Comment (normanj-bitquill) [reply testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java:4597]: Added more tests for both `TO_DATE` and `TO_TIMESTAMP`.

**Comment by mihaibudiu** [comment]
In order to keep track of JIRA/github, we require the JIRA title to be the same as the PR title, and the commit message.
Can you please make sure they match?
The JIRA title is good if you remove "TO_CHAR" from it.

**Review by mihaibudiu — APPROVED** [review]

**Comment by normanj-bitquill** [comment]
> In order to keep track of JIRA/github, we require the JIRA title to be the same as the PR title, and the commit message. Can you please make sure they match? The JIRA title is good if you remove "TO_CHAR" from it.

@mihaibudiu I don't have permission to change the title in Jira or to assign the ticket to myself. Can I get the necessary permissions?

**Comment by mihaibudiu** [comment]
I edited the JIRA title. You shoud change the commit and PR titles.
But if you have a JIRA account and you filed the issue I would expect you to be able to edit it... I don't know much about JIRA, though. Don't worry about assigning it.

**Comment by normanj-bitquill** [comment]
@mihaibudiu Thanks for changing the title in Jira. I have updated the commit message and title of this PR. I'll sort out my Jira permissions.

**Pull request merged**